subsys/modbus/modbus_serial.c

```diff
@@ -527,8 +527,10 @@ int modbus_serial_rx_adu(struct modbus_context *ctx)
 		return -ENOTSUP;
 	}
 
+	unsigned int key = irq_lock();
 	cfg->uart_buf_ctr = 0;
 	cfg->uart_buf_ptr = &cfg->uart_buf[0];
+	irq_unlock(key);
```

Comment:
IIRC, RX interrupt should be disabled at this point, there should be modbus_serial_rx_disable() called before. What platform/driver are you using to reproduce it?

Reply:
You are absolutely correct, I missed the `modbus_serial_rx_disable()` in modbus_rx_handler and thus my analysis of the problem is incorrect. (The problem is that we are seeing intermittent modbus-adjacent memory corruption during high noise testing.)

I will close this PR and keep investigating.